enhancements/sig-architecture/6-placements/README.md

```diff
@@ -0,0 +1,523 @@
+
+### Workflows
+
+#### Workflow 1: Placement Reconciliation
+1. If the placement has old placementdicisions, fetch them all and get a slice of old matched clusters;
+2. Evaluate each cluster predicates (see Workflow 2 below for more details) and get the union of all predicates result;
+3. Enforce the affinity constrains on the union and remove clusters if necessary;
+4. If the placement has no old placementdicisions yet, create placementdicisions;
+5. Otherwise, update the existing placementdicisions, and add/remove placementdicisions if necessary.
+
+#### Workflow 2: Evaluate a cluster predicate
+1. If `ClusterName` is specified, return cluster if it is exists;
+2. If `ClusterSets` is not empty, get clusters belongs to intersection of `ClusterSets` and the ManagedClusterSets bound to the placement namespace;
+3. Otherwise, get clusters from the ManagedClusterSets bound to the placement namespace;
+4. If `RequiredDuringSchedulingRequiredDuringExecution` is specified, filter clusters with label/claim selectors; 
+5. Enforce the affinity constrains on clusters if exist;
+6. If `NumberOfClusters` is not specified, return all matched clusters;
+7. Otherwise, get the intersection of old matched clusters and new matched clusters;
+8. If `NumberOfClusters` is equal to the size of the intersection, return clusters in the intersection;
+9. if `NumberOfClusters` is less than the size of the intersection, select random clusters from the intersection and return;
+10. Otherwise, return clusters in the intersection with some random additional clusters from the remaining of the new matched clusters;
+
+**Note:** The old matched clusters are preferred during the predicate evaluation, which makes the placement decisions as stable as possible.
+
+### Open Questions
+
+1. The API group of the new placement APIs could be `cluster.open-cluster-management.io` or `scheduling.open-cluster-management.io`. Which one is better?
+
+### Test Plan
+
+- Unit tests will cover the functionality of the controllers.
+- Unit tests will cover the new APIs.
+- Integration tests will cover all user stories.
+- And e2e tests will cover the following cases.
+  - Create/update/delete a placement;
+  - ManagedCluster labels/claims changes;
+  - A ManagedClusterSet is bound/unbound to/from the placement namespace;
+  - A ManagedCluster is added/removed to/from a bound ManagedClusterSet;
+
+### Graduation Criteria
+
+#### Alpha
+1. The new APIs is reviewed and accepted;
+2. Implementation is completed to support the functionalities;
+3. Develop test cases to demonstrate that the above user stories work correctly;
+
+#### Beta
+1. The new APIs is able to cover all the capabilities of the current placementrule APIs;
+2. Conversion webhook is provided to migrate existing placement objects to the new placement APIs;
+3. At lease one component uses the new APIs;
+4. Support selecting clusters belong to no ManagedClusterSet;
+5. Support tolerations/taints (user story 8);
```

Comment:
Is this a goal or a non-goal? In the beginning of this document it's nak'd by being in the list of non-goals

Reply:
I moved them to Goals section. And we will consider to support them in beta.